runtime/convert.go

```diff
@@ -85,3 +86,23 @@ func Duration(val string) (*duration.Duration, error) {
 	err := jsonpb.UnmarshalString(val, r)
 	return r, err
 }
+
+// Enum converts the given string into an int32 that should be type casted into the
+// correct enum proto type.
+func Enum(val string, enumValMap map[string]int32) (int32, error) {
+	e, ok := enumValMap[val]
```

Comment:
This lookup means we define the correct enum values as the strings generated by the go Protobuf implementation. Is this appropriate? It'll be things like `MyEnum_FIRST_VALUE` etc. Is there some way we can use the values from the protofile directly (i.e. `FIRST_VALUE`). I like having the protofile content as the source of truth.

Reply:
Not sure I follow. The convert Enum function will use the name->index map that is generated by the proto compiler, see e.g. [grpc-gateway/examples/proto/examplepb/a_bit_of_everything.pb.go]( https://github.com/maros7/grpc-gateway/blob/enum-path-parameter/examples/proto/examplepb/a_bit_of_everything.pb.go#L48-L51). That map is provided to the convert Enum function when conversion from path parameters to protobuf is done, see e.g. [grpc-gateway/examples/proto/examplepb/a_bit_of_everything.pb.gw.go](https://github.com/maros7/grpc-gateway/blob/enum-path-parameter/examples/proto/examplepb/a_bit_of_everything.pb.gw.go#L216-L253). So not sure exactly what you mean.

Reply:
Ah, my bad, the values aren't prefixed. Carry on!